src/object/srv_obj.c

```diff
@@ -1431,8 +1483,12 @@ obj_local_rw_internal(crt_rpc_t *rpc, struct obj_io_context *ioc,
 			goto out;
 		}
 		if (recov_lists != NULL) {
-			daos_recx_ep_list_set_ep_valid(recov_lists,
-						       orw->orw_nr);
+			daos_epoch_t	eph_boundry;
+
+			eph_boundry = obj_fetch_from_snapshot(orw, ioc) ?
+				      0 : ioc->ioc_coc->sc_ec_agg_eph_boundry;
```

Comment:
I don't quite follow this, shouldn't we use the snapshot epoch for degraded fetch?

Reply:
if passing 0 to daos_recx_ep_list_set, the original recv_list's epoch will not be changed, so keep as original parity ext's epoch. But using snapshot epoch is better.

Another problem is, seems need to consider the case of snapshot epoch is higher than vos agg epoch boundary, as the following VOS agg possibly change the result of fetching from snapshot epoch.

Reply:
Why using the original parity epoch for snapshot fetch is ok? If EC aggregation merged parity extent to low epoch but VOS aggregation on data shard didn't start, using the parity epoch would fetch wrong data on data shard?